src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Forms/FormsWithFiles.razor

```diff
@@ -0,0 +1,57 @@
+﻿@page "/forms/with-files"
+@using Microsoft.AspNetCore.Components.Forms
+
+<h2>Forms With Files</h2>
+
+<EditForm Model="Model" method="POST" OnValidSubmit="DisplaySuccess" FormName="fileForm" enctype="multipart/form-data">
+    <ValidationSummary />
+    <div>
+        <label for="Model.ProfilePicture">
+            Profile Picture:
+            <InputFile @bind-Value="Model.ProfilePicture" name="Model.ProfilePicture" />
```

Comment:
Can you copy the logic from https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web/src/Forms/InputBase.cs#L199-L221 on to InputFile so that it doesn't require specifying the name manually?

Reply:
It'll need to be added https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web/src/Forms/InputFile.cs#L62

Reply:
I ended up keeping this change out of this PR to keep the scope smaller.

Reply:
That's fair. I'll file an issue for it.

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/50728 to track it